File: Neetcode/RemoveNthfromEndLL.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        if head.next is None and n == 1:
            return None
        elif head.next is None:
            return head
        nodecount = 0
        h = head
        while h is not None:
            nodecount += 1
            h = h.next
        if nodecount == n:
            return head.next
        delnode = nodecount - n
        h = head
        i = 0
        while h is not None and i < delnode:
            if i+1 == delnode:
                h.next = h.next.next
            h = h.next
            i += 1
        return head

File: Neetcode/RemoveNthfromEndLL.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int num = 0;
        ListNode* h = head;
        while (h != nullptr){
            num += 1;
            h = h->next;
        }
        num = num - n;
        h = head;
        if (num){
            while (num-1){
                num -= 1;
                h = h->next;
            }
        }
        else{
            return head->next;
        }
        h->next = h->next->next;
        return head;
    }
};
